### PR TITLE
fix(certificatemanager): domainName not checked for length

### PR DESCRIPTION
**Pull request by josephedward**
## fix(AWS-CertificateManager) cap domain name to 64 characters
*fix for*: https://github.com/aws/aws-cdk/issues/21777
*motivation*: if a domainName that is longer than 64 characters is provided, certification creation will fail - this implements a check for string length 

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by peterwoodworth — DISMISSED** [review]
You have a large number of changes in the `eks-legacy` package for some reason, I don't think there should be any changes here.

**Comment by josephedward** [comment]
Thanks @peterwoodworth, I had a merge commit that included a bunch of upstream work, was not sure if it was failing because of that, but I reverted.

File: packages/@aws-cdk/aws-certificatemanager/lib/certificate.ts [hunk 1]
```diff
@@ -227,6 +227,11 @@ export class Certificate extends CertificateBase implements ICertificate {
       }
     }
 
+    //check if domain name is less than 64  characters
+    if (props.domainName.length > 64) {
+      throw new Error('Domain name must be less than 64 characters');
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-certificatemanager/lib/certificate.ts:232]: The code indicates that the domain name must be 64 characters or less, not less than 64 characters. Please make sure the code and error message are consistent.

File: packages/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts [hunk 2]
```diff
@@ -86,6 +86,10 @@ export class DnsValidatedCertificate extends CertificateBase implements ICertifi
     this.region = props.region;
 
     this.domainName = props.domainName;
+    //check if domain name is less than 64  characters
+    if (this.domainName.length > 64) {
+      throw new Error('Domain name must be less than 64 characters');
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts:91]: Same comment as above.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution! This looks mostly good, I just have a couple comments inline.

File: packages/@aws-cdk/aws-certificatemanager/lib/certificate.ts (hunk 1 above)
Comment (josephedward) [reply packages/@aws-cdk/aws-certificatemanager/lib/certificate.ts]: Off by one strikes again! Good catch, thanks @TheRealAmazonKendra :)

File: packages/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts [hunk 3]
```diff
@@ -86,6 +86,10 @@ export class DnsValidatedCertificate extends CertificateBase implements ICertifi
     this.region = props.region;
 
     this.domainName = props.domainName;
+    //check if domain name is less than 64  characters
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts:89]:
```suggestion
    // check if domain name is 64 characters or less
```

**Review by TheRealAmazonKendra — DISMISSED** [review]
I updated the comments and spacing slightly because it was too much of a nit to ask you to do. Approving this.

**Comment by TheRealAmazonKendra** [comment]
Don't use the merge button when performing merges from `main`. Please use @Mergifyio update.

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**